src/jit/lsra.cpp

```diff
@@ -6390,6 +6388,73 @@ regNumber LinearScan::assignCopyReg(RefPosition* refPosition)
     return allocatedReg;
 }
 
+//------------------------------------------------------------------------
+// isAssigned: This is the function to check if the given RegRecord has an assignedInterval
+//             regardless of lastLocation.
+//             So it would be call isAssigned() with Maxlocation value.
+//
+// Arguments:
+//    regRec       - The RegRecord to check that it is assigned.
+//    newRegType   - There are elements to judge according to the upcoming register type.
+//
+// Return Value:
+//    Returns true if the given RegRecord has an assignedInterval.
+//
+// Notes:
+//    There is the case to check if the RegRecord has an assignedInterval regardless of Lastlocation.
+//
+bool LinearScan::isAssigned(RegRecord* regRec ARM_ARG(RegisterType newRegType))
+{
+    return isAssigned(regRec, MaxLocation ARM_ARG(newRegType));
```

Comment:
This is not the most efficient way to do this, as it will always check `lastLocation` against `MaxLocation`, and that check will not be easily optimized away by the C++ compiler. It would be better to make this an entirely separate method, or to add a boolean argument to indicate whether to do the extra checking (the latter would be more likely to enable the C++ compiler to do the necessary optimization). I'm OK with filing an issue for this. @pgavlin - your thoughts?